File: Server/Server/Network/PacketHandler/ILdpObservable.cs
using Server.Network.PacketSender;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Server.Network.PacketHandler
{
    interface ILdpObservable
    {
        void AddListener(ILdpPacketHandler listener);
        void RemoveListener(ILdpPacketHandler listener);
        void RemoveListeners();
        void NotifyToAllListeners(LdpPacket packet, ILdpPacketSender channel);
    }
}

File: AndroidClient/LdpAndroidClient/app/src/main/java/com/ldp/androidclient/network/packet_handler/ILdpObservable.java
package com.ldp.androidclient.network.packet_handler;


import com.ldp.androidclient.protocol.*;

public interface ILdpObservable {
    void addListener(ILdpPacketHandler listener);
    void removeListener(ILdpPacketHandler listener);
    void removeListeners();
    void notifyToAllListeners(LdpProtocol.LdpPacket packet);
}
